Use a locale that is already in libc-bin, so no need to use B-D on locales.

Upstream issue:
https://github.com/neomutt/neomutt/issues/1088

--- a/test/rfc2047.c
+++ b/test/rfc2047.c
@@ -53,7 +53,10 @@
 
 void test_rfc2047(void)
 {
-  setlocale(LC_ALL, "en_US.UTF-8");
+  if (!TEST_CHECK(setlocale(LC_ALL, "C.UTF-8") != NULL)) {
+    TEST_MSG("Cannot set locale to C.UTF-8");
+    return;
+  }
   Charset = "utf-8";
 
   for (size_t i = 0; i < mutt_array_size(test_data); ++i)
